src/domain/organizations/organizations-safe.repository.spec.ts

```
@@ -0,0 +1,679 @@
+          address: getAddress(faker.finance.ethereumAddress()),
+        }),
+        { count: { min: 2, max: 5 } },
+      );
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      await orgSafesRepo.create({
+        organizationId: orgId,
+        payload: orgSafes,
+      });
+      const orgSafeBefore = await orgSafesRepo.findByOrganizationId(orgId);
+      expect(orgSafeBefore).toHaveLength(orgSafes.length);
+
+      await orgSafesRepo.delete({
+        organizationId: orgId,
+        payload: orgSafes,
+      });
+
+      await expect(
+        orgSafesRepo.findOrFail({
+          where: { organization: { id: orgId } },
+        }),
+      ).rejects.toThrow('Organization has no Safes.');
+    });
+
+    it('should throw NotFoundException if provided OrganizationSafe is not found', async () => {
+      const org = await dbOrgRepo.insert({
+        status: faker.helpers.arrayElement(
+          getStringEnumKeys(OrganizationStatus),
+        ),
+        name: faker.word.noun(),
+      });
+      const orgId = org.identifiers[0].id as Organization['id'];
+      const chainId = faker.string.numeric();
+      await dbOrgSafeRepo.insert({
+        chainId,
+        address: getAddress(faker.finance.ethereumAddress()),
+        organization: {
+          id: orgId,
+        },
+      });
+
+      await expect(
+        orgSafesRepo.delete({
+          organizationId: orgId,
+          payload: [
+            {
+              chainId,
+              address: getAddress(faker.finance.ethereumAddress()),
+            },
+          ],
+        }),
+      ).rejects.toThrow('Organization has no Safes.');
+    });
+
+    it('should throw NotFoundException if none of the provided OrganizationSafes is found', async () => {
```

I would additionally check the behavior when any of the provided OrganizationSafes is not found.

---

What would be the expected behavior in that case? Should we fail if one or more of the provided OrganizationSafes are not found? @PooyaRaki @iamacook 

The current implementation only fails when **all** are non-existent, because it relies in `findOrFail`, and this method has the inner logic:

```
if (organizationSafes.length === 0) {
  throw new NotFoundException('Organization has no Safes.');
}
```

---

The current implementation fails fast i.e. as soon as one of the Safes in request is not found the service throws.

---

The current implementation fails if no org is found, even if one org is found it tries to delete that one org.